modules/apps/27-interchain-accounts/controller/keeper/account.go

```diff
@@ -9,45 +11,56 @@ import (
+// - The MsgChannelOpenInitResponse response is returned.
+// - An error is also returned if the port identifier is already in use. Gaining access to interchain accounts whose channels
 // have closed cannot be done with this function. A regular MsgChannelOpenInit must be used.
-func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, connectionID, owner, version string) error {
+func (k Keeper) registerInterchainAccount(ctx sdk.Context, connectionID, owner, version string) (channeltypes.MsgChannelOpenInitResponse, error) {
 	portID, err := icatypes.NewControllerPortID(owner)
 	if err != nil {
-		return err
+		return channeltypes.MsgChannelOpenInitResponse{}, err
 	}
 
 	// if there is an active channel for this portID / connectionID return an error
 	activeChannelID, found := k.GetOpenActiveChannel(ctx, connectionID, portID)
 	if found {
-		return sdkerrors.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s on connection %s for owner %s", activeChannelID, portID, connectionID, owner)
+		return channeltypes.MsgChannelOpenInitResponse{}, sdkerrors.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s on connection %s for owner %s", activeChannelID, portID, connectionID, owner)
 	}
 
 	switch {
 	case k.portKeeper.IsBound(ctx, portID) && !k.IsBound(ctx, portID):
-		return sdkerrors.Wrapf(icatypes.ErrPortAlreadyBound, "another module has claimed capability for and bound port with portID: %s", portID)
+		return channeltypes.MsgChannelOpenInitResponse{}, sdkerrors.Wrapf(icatypes.ErrPortAlreadyBound, "another module has claimed capability for and bound port with portID: %s", portID)
 	case !k.portKeeper.IsBound(ctx, portID):
 		cap := k.BindPort(ctx, portID)
 		if err := k.ClaimCapability(ctx, cap, host.PortPath(portID)); err != nil {
-			return sdkerrors.Wrapf(err, "unable to bind to newly generated portID: %s", portID)
+			return channeltypes.MsgChannelOpenInitResponse{}, sdkerrors.Wrapf(err, "unable to bind to newly generated portID: %s", portID)
 		}
 	}
 
 	msg := channeltypes.NewMsgChannelOpenInit(portID, version, channeltypes.ORDERED, []string{connectionID}, icatypes.PortID, icatypes.ModuleName)
 	handler := k.msgRouter.Handler(msg)
-
 	res, err := handler(ctx, msg)
 	if err != nil {
-		return err
+		return channeltypes.MsgChannelOpenInitResponse{}, err
 	}
 
 	// NOTE: The sdk msg handler creates a new EventManager, so events must be correctly propagated back to the current context
 	ctx.EventManager().EmitEvents(res.GetEvents())
 
-	return nil
+	firstMsgResponse := res.MsgResponses[0]
+	channelOpenInitResponse, ok := firstMsgResponse.GetCachedValue().(*channeltypes.MsgChannelOpenInitResponse)
+	if !ok {
+		return channeltypes.MsgChannelOpenInitResponse{}, sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "failed to covert %s message response to %s", reflect.TypeOf(firstMsgResponse.GetCachedValue()), reflect.TypeOf(&channeltypes.MsgChannelOpenInitResponse{}))
```

Comment:
What is the difference if you use `%T`?

Reply:
I didn't know about the %T formatting option! Looks like all we need 👍